### ISSUE TITLE
Can't test "file" type inputs.

### ISSUE THREAD

**Issue by mveraldi-cisco** [testing-library/react-testing-library#1286]
### Environment
- `@testing-library/react` version: 14.2.1
- Testing Framework and version:
  `react-scripts` version: 5.0.1
- DOM Environment:
  `jsdom` from `react-scripts`

### Reproduction:

Minimal reproduction on CodeSandbox: [here](https://codesandbox.io/p/devbox/testing-library-input-file-issue-8zq2mj?file=%2Fpackage.json%3A23%2C43&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cltwpghcr00063b6k9jrnbrfu%2522%252C%2522sizes%2522%253A%255B60%252C40%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cltwpghcr00023b6k66mjb7yx%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cltwpghcr00033b6kwa7nx2vt%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cltwpghcr00053b6kfn444794%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cltwpghcr00023b6k66mjb7yx%2522%253A%257B%2522id%2522%253A%2522cltwpghcr00023b6k66mjb7yx%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltwpoi1p00023b6jf416b2kl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A11%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A11%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252F__tests__%252Fhello.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cltwr9zjw00023b6j2mmdo2u1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A23%252C%2522startColumn%2522%253A43%252C%2522endLineNumber%2522%253A23%252C%2522endColumn%2522%253A43%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltwr9zjw00023b6j2mmdo2u1%2522%257D%252C%2522cltwpghcr00053b6kfn444794%2522%253A%257B%2522id%2522%253A%2522cltwpghcr00053b6kfn444794%2522%252C%2522activeTabId%2522%253A%2522cltwpx4cr00sr3b6jgfsde7je%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltwpghcr00043b6kd6x7y298%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cltwpsd1w00753b6jtmcsr6io%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522cltwpx4cr00sr3b6jgfsde7je%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522cltwpghcr00033b6kwa7nx2vt%2522%253A%257B%2522id%2522%253A%2522cltwpghcr00033b6kwa7nx2vt%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cltwps9vz005f3b6je9f1js1k%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522cltwpszvw00cz3b6jibf3bjax%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cltwpszwv000pdbgmb86h3uhp%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltwpszvw00cz3b6jibf3bjax%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### What you did:

I need to assert that my api is receiving the correct files. In this minimal reproduction it might not make sense because it's like I am testing DOM, but in my real world scenario the form is more complex and in that context this test makes sense.

### What happened:

when I call 
`expect(api).toHaveBeenCalledWith({name: "hello.txt", file: file})`
I receive `{name: "", file: {}}`.

Basically I am unable to assert what file was passed in the function.

Thank you.
